contrib/babelfishpg_tsql/src/pl_exec.c

```diff
@@ -6410,6 +6410,28 @@ exec_assign_value(PLtsql_execstate *estate,
 										   var->datatype->typoid,
 										   var->datatype->atttypmod);
 
+				/* Special handling when target variable is babelfish GUC */
+				if(var->is_babelfish_guc)
+				{	
+					StringInfoData buf;
+					if (isNull)
+						ereport(ERROR,
+								(errcode(ERRCODE_NULL_VALUE_NOT_ALLOWED),
+								errmsg("Invalid argument for SET %s. Must be a non-null value.",
+										var->refname)));
+
+					initStringInfo(&buf);
```

Comment:
buf.data which is a pointer, was allocated 1024 bytes, where are freeing it?
Also use StringInfo instead of StringInfoData

Reply:
addressed in next commit